buildings_tools/src/buildings_tools/Building.java

```diff
@@ -265,7 +287,80 @@ private Node getAddressNode() {
         return nodes.get(0);
     }
 
-    public Way create() {
+    public Way createCircle() {
+        DataSet ds = MainApplication.getLayerManager().getEditDataSet();
+        Collection<OsmPrimitive> selectedPrimitives = ds.getAllSelected();
+        ds.clearSelection();
+
+        if (len == 0)
+            return null;
+        final boolean[] created = new boolean[2];
+        final Node[] nodes = new Node[2];
+        for (int i = 0; i < 2; i++) {
+
+            Node n = findNode(en[i]);
+            if (n == null) {
+                nodes[i] = new Node(eastNorth2latlon(en[i]));
+                created[i] = true;
+            } else {
+                nodes[i] = n;
+                created[i] = false;
+            }
+            if (nodes[i].getCoor().isOutSideWorld()) {
+                JOptionPane.showMessageDialog(Main.parent,
+                        tr("Cannot place building outside of the world."));
+                return null;
+            }
+        }
+        Way w = new Way();
+        w.addNode(nodes[0]);
+        w.addNode(nodes[1]);
+
+        Collection<Command> addNodesCmd = new LinkedList<>();
+
+        for (int i = 0; i < 2; i++) {
+            if (created[i]) {
+                AddCommand addNode = new AddCommand(ds, nodes[i]);
+                addNodesCmd.add(addNode);
+            }
+        }
+
+        if (addNodesCmd.size() > 0) {
+            Command addNodes = new SequenceCommand(tr("Add nodes for building"), addNodesCmd);
+            Main.main.undoRedo.add(addNodes);
+        }
+
+        // Nodes must be selected for create circle action
+        for (int i = 0; i < 2; i++) {
+            if (created[i]) {
+                ds.addSelected(nodes[i]);
+            }
+        }
+
+        CreateCircleAction action = new CreateCircleAction();
+        action.setEnabled(true);
+        action.actionPerformed(null);
+
+        ds.clearSelection();
+        ds.addSelected(selectedPrimitives);
+
+        int lastWayIndex = ds.getWays().size() - 1;
+        List<Way> ways = new ArrayList(ds.getWays());
+        Comparator<Way> comparator = new Comparator<Way>() {
+            @Override
+            public int compare(Way left, Way right) {
+                return Long.compare(right.getUniqueId(), left.getUniqueId());
+            }
+        };
+        Collections.sort(ways, comparator);
```

Comment:
Seems like there is already a compareTo method on Way.java.
>    @Override
    public int compareTo(OsmPrimitive o) {
        if (o instanceof Relation)
            return 1;
        return o instanceof Way ? Long.compare(getUniqueId(), o.getUniqueId()) : -1;
    }

Then I realized your comparator flips the arguments which I guess is necessary because the newer features have a smaller negative number as the id?  If so we couldn't we just reverse the sort by the "natural" order (without a custom comparator)?